syntax = "proto3";

package masterworker;

service MasterWorker{
	  
	rpc mapReduceQuery (MasterQuery) returns (WorkerReply) {}
	
}


message MapperQuery {
	Shard shard = 1;
}

message Shard {
	int32 id = 1;
	string filename = 2;
	int32 startByte = 3;
	int32 endByte = 4;
}

message ReducerQuery {
	FileLocations locations = 1;
}

message WorkerReply {
	FileLocations locations = 1;
}

message FileLocations {
	repeated string filename = 1;
}

message MasterQuery {
	MapperQuery mapperQuery = 1;
	ReducerQuery reducerQuery = 2;
	int32 type = 3;
}

